File: encode/encode.cpp
#include <iostream>
#include <string>
#include <stdio.h>

using namespace std;

int main(){
        // put your video name here
        string input="./samplevideo.mp4"; // video file
        string outvideoname = "test.mkv"; // output video name
        string cmd = "/opt/encode.sh -i "+input+" -o "+outvideoname+" -v h264";
        system(cmd.c_str());
        return 0;
}
 

File: encode/encode.py
!/usr/bin/env python3

'''
Simple example to demonstrate HW accelerated encoding using python on BrainyPi.

Note: Do not use this code in production.
'''

import subprocess

# Put input video file name here
inputfile="./samplevideo.mp4"

# Put output video file name here
outvideoname = "test.mkv"

a= subprocess.call(['/opt/encode.sh','-i',inputfile,'-o',outvideoname,'-v','h264'])  
